### PR TITLE
Feat(eos_cli_config_gen): add support for FlexRoute compression and expansion of prefixes

### LINKED ISSUE #1620: FlexRoute: add support for compression and expansion of prefixes
### Enhancement summary

According to https://aristanetworks.force.com/AristaCommunity/s/article/inet-edge-config

### Which component of AVD is impacted

eos_cli_config_gen

### Use case example

You can enable FlexRoute compression and expansion of prefixes.  While these commands can be customized for different prefix lengths, Arista has included pre-defined 'internet' profiles for both IPv4 and IPv6 based on the current contents of the Internet routing table.

#### IPv4
```yaml
ip_hardware:
  fib:
    optimize:
      prefix_profile: < internet | urpf-internet >
```

#### IPv6
```yaml
ipv6_hardware:
  fib:
    optimize:
      prefix_profile: < internet >
```

### Describe alternatives you have considered

_No response_

### Additional context

PR incoming

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by c-po**
## Change Summary

You can enable FlexRoute compression and expansion of prefixes. While these commands can be customized for different prefix lengths, Arista has included pre-defined 'internet' profiles for both IPv4 and IPv6 based on the current contents of the Internet routing table.

## Related Issue(s)

Fixes #1620

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes

```yaml
ip_hardware:
  fib:
    optimize:
      prefix_profile: < internet | urpf-internet >
```

```yaml
ipv6_hardware:
  fib:
    optimize:
      prefix_profile: < internet >
```

## How to test

Molecule

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/ip-hardware.j2 [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+#### Optimize
+{%       if ip_hardware.fib.optimize.prefix_profile is arista.avd.defined %}
+
+##### Prefix Profile
+
+| Profile |
+| ------- |
+| {{ ip_hardware.fib.optimize.prefix_profile }} |
+
+### IP Hardware Device Configuration
```
Comment (tgodaA) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/ip-hardware.j2]: line 18-22 are independent of the prefix_profile (Currently this hardware device config will be printed out only if prefix_profile is defined). Let's say in future someone adds a new knob under FIB. So please keep this section under ip_hardware if condition.

Comment (tgodaA) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/ip-hardware.j2]: Similarly for the other template.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/ip-hardware.j2 [hunk 2]
```diff
@@ -0,0 +1,26 @@
+{% if ip_hardware is arista.avd.defined %}
+
+## IP Hardware
+{%   if ip_hardware.fib is arista.avd.defined %}
```
Comment (tgodaA) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/ip-hardware.j2]: Indentation should be 4 spaces(you provided 2 spaces at every next level). There is a CI check failing due to indentation. Please make sure indentation at all if statements.

Run ```pre-commit run -a``` before pushing any changes to make sure you catch these errors locally.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/ip-hardware.j2 [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{# eos - IP Hardware #}
+{% if ip_hardware is arista.avd.defined %}
+!
+{%   if ip_hardware.fib.optimize.prefix_profile is arista.avd.defined %}
```
Comment (tgodaA) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/ip-hardware.j2]: indentation error. Should be 4 spaces with respect to outer if. But you gave only 2 spaces. Please check in all templates.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/ip-hardware.j2 (hunk 2 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/ip-hardware.j2]: these will only be caught by our `j2lint` library.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/ip-hardware.j2 [hunk 4]
```diff
@@ -0,0 +1,7 @@
...
+ip hardware fib optimize prefixes profile {{ ip_hardware.fib.optimize.prefix_profile }}
```
Comment (xaviramon) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/ip-hardware.j2]: I think we should change the data model to match the EOS syntax:
```
smv031(config)#ip hardware fib optimize ?
  disable-vrf    Disable fib optimizations in Vrf
  prefix-length  Length of the prefix in bits
  prefixes       Optimize prefixes
  vrf            Optimize IPv4 routes for specific VRF(s)
  <cr>
```
So it should be:
```
ip_hardware:
  fib:
    optimize:
      prefixes:
        profile: < internet | urpf-internet >

**Comment by xaviramon** [comment]
These commands should be included into the IP-routing eos templates, because the running config generated in the device is as follows:
```
ip routing
no ip icmp redirect
ip hardware fib optimize prefixes profile internet
no ip routing vrf MGMT
!
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/ip-hardware.j2 (hunk 4 above)
Comment (c-po) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/ip-hardware.j2]: Aggreed and changed

**Review by ClausHolbechArista — APPROVED** [review]
LGTM.
It is a bit clunky that these commands are in the ip-routing template, but I understand why, when you look at the config ordering with vrfs etc.

**Review by tgodaA — APPROVED** [review]
LGTM

**Pull request merged**